### PR TITLE
feature/ja-61_RepositoryBase

### PR DESCRIPTION
**Pull request by skrawus**

### CODE REVIEW

File: TutorLizard.BusinessLogic/Data/RepositoryBase.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            string filePath = Path.Combine(_filePath.Split('/'));
+            string fullPath;
+            if (Path.IsPathRooted(filePath))
+                fullPath = filePath;
+            else
+                fullPath = Path.Combine(AppContext.BaseDirectory, filePath);
+
+            if (!File.Exists(fullPath))
+                _data = new List<T>();
```
Comment (Zjyslav) [inline TutorLizard.BusinessLogic/Data/RepositoryBase.cs]: Replace this line (35) with `return;`
We changed return to assignment, but it's actually important to return here (but the value is not important).
`_data` initializes itself, so there's no need to assign empty list to it.

**Review by Zjyslav — CHANGES_REQUESTED** [review]
I looked into the way we could implement getting `filePath` from configuration and if we want to do it in the derived classes, specifying constructor in the base class hurts us, because it's impossible to set the `_filePath` in derived constructor before the base constructor calls `LoadFromJson()` that uses this path.

I think that the most flexible approach would be to remove the constructor and replace:
`private string _filePath;`
with
`protected abstract string FilePath { get; }`

This will force any derived class to implement `FilePath`, but allow to do it either specify the value inline with the implementation or in the constructor. Each implementation would  have to call `LoadFromJson()`, but it would have the freedom to decide when it's ready to to so.
Implementation could look like this:
```
public class AdRepository : RepositoryBase<Ad>
{
    protected override string FilePath => "Data/Ads.json";
    public AdRepository()
    {
        LoadFromJson();
    }
}
```

or: 
```
public class UserRepository : RepositoryBase<User>
{
    protected override string FilePath { get; }

    public UserRepository(IConfiguration config)
    {
        string section = "DataFilePaths:Users";
        string? path = config.GetValue<string>(section);
        if (string.IsNullOrWhiteSpace(path))
            throw new Exception($"Configuration section missing: {section}");
        FilePath = path;
        LoadFromJson();
    }
}
```

File: TutorLizard.BusinessLogic/Data/RepositoryBase.cs (hunk 1 above)
Comment (skrawus) [reply TutorLizard.BusinessLogic/Data/RepositoryBase.cs]: This forced me to make few more changes (so the code looks like in DA now)

**Comment by skrawus** [comment]
The requested changes are pushed. :)

File: TutorLizard.BusinessLogic/Data/RepositoryBase.cs (hunk 1 above)
Comment (Zjyslav) [reply TutorLizard.BusinessLogic/Data/RepositoryBase.cs]: If you did `return;` instead of `return new List<T>();` you wouldn't have to make these changes.

**Review by Zjyslav — APPROVED** [review]

**Pull request merged**